### PR TITLE
Update to version 10.4 and use tarball istead of repo

### PR DESCRIPTION
**Pull request by prusse-martin**

### CODE REVIEW

**Comment by prusse-martin** [comment]
From https://conda-forge.org/docs/meta.html#building-noarch-packages
> ... since it’s only built once (on CircleCI) ...

I am pretty confident we can set the env var that way.

Note that I added `conda-devenv --version` to the tests commands soo we can check if it is working properly with circle ci and conda forge infrastructure.

**Comment by prusse-martin** [comment]
Frrom the [build history](https://circleci.com/gh/conda-forge/conda-devenv-feedstock/41#tests/containers/0):
> \+ /bin/bash -x -e /feedstock_root/build_artefacts/conda-devenv_1538070256336/test_tmp/run_test.sh
> \+ conda-devenv --version
> conda-devenv version 1.0.4

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-  git_rev: {{ version }}
+  url: https://github.com/ESSS/conda-devenv/archive/{{ version }}.tar.gz
+  sha256: a0a88f7d482ec6293aee52a3026055e32a8719ec828f0496fc9b5c084c7ab33f
 
 build:
-  number: 1
-  script: python -m pip install . --no-deps --ignore-installed
+  number: 0
+  script:
+    - export SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
```
Comment (nicoddemus) [inline recipe/meta.yaml:15]: @ocefpaf what do you think about this? Is this something acceptable in general for `noarch` packages?

Comment (ocefpaf) [reply recipe/meta.yaml:15]: We are running on Linux and, as long as that does not bake any Linux specific stuff on the built package, that is OK for `noarch`.

Comment (nicoddemus) [reply recipe/meta.yaml:15]: Thanks for the feedback!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+    - python -m pip install . --no-deps --ignore-installed
```
Comment (nicoddemus) [inline recipe/meta.yaml:16]: Oh just realized that we should update this to:

```
"{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```

But we can do that the next time we update.

**Pull request merged**